awx/ui_next/src/components/PaginatedDataList/ToolbarDeleteButton.jsx

```diff
@@ -66,20 +80,44 @@ function ToolbarDeleteButton({
   pluralizedItemName,
   errorMessage,
   onDelete,
+  deleteDetailsRequests,
   warningMessage,
+  deleteMessage,
   i18n,
   cannotDelete,
 }) {
   const { isKebabified, onKebabModalChange } = useContext(KebabifiedContext);
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [deleteDetails, setDeleteDetails] = useState({});
 
+  const deleteMessages = [warningMessage, deleteMessage].filter(
+    message => message
+  );
+
+  const [deleteMessageError, setDeleteMessageError] = useState();
   const handleDelete = () => {
     onDelete();
     toggleModal();
   };
 
-  const toggleModal = () => {
-    setIsModalOpen(!isModalOpen);
+  const toggleModal = async isOpen => {
```

Comment:
We can wrap this function on `useCallback` to avoid to be recreated every re-render.